dev/com.ibm.ws.jndi/src/com/ibm/ws/jndi/internal/WASInitialContextFactoryBuilder.java

```diff
@@ -31,16 +31,22 @@
     public InitialContextFactory createInitialContextFactory(Hashtable<?, ?> environment)
                     throws NamingException {
 
-        String icfFactory = (String) environment.get(Context.INITIAL_CONTEXT_FACTORY);
+        final String icfFactory = (String) environment.get(Context.INITIAL_CONTEXT_FACTORY);
         InitialContextFactory icf = null;
 
         if (icfFactory != null) {
-            try {
-                Class<?> clazz = Class.forName(icfFactory, true, getClassLoader());
-                icf = (InitialContextFactory) clazz.newInstance();
-            } catch (Exception e) {
-                //auto FFDC
-            }
+            icf = AccessController.doPrivileged(new PrivilegedAction<InitialContextFactory>() {
+                @Override
+                public InitialContextFactory run() {
+                    try {
+                        Class<?> clazz = Class.forName(icfFactory, true, getClassLoader());
```

Comment:
This needs to be `false` to prevent initialisation within the `doPrivileged` block.